backend/app/services/posting_service.rb

```diff
@@ -65,14 +65,71 @@ def prefill(user:)
         applicant = Applicant.find_by(utorid: utorid)
         return { utorid: utorid } unless applicant
 
-        {
+        # This is the basic data we can get if the applicant already exists
+        data = {
             utorid: utorid,
             student_number: applicant.student_number,
             first_name: applicant.first_name,
             last_name: applicant.last_name,
             email: applicant.email,
             phone: applicant.phone
         }
+
+        existing_application = Application.find_by(posting: @posting)
+        if existing_application
+            application_service =
+                ApplicationService.new application: existing_application
+            data.merge! application_service.prefilled_data
+        else
+            # Find out if they have previous contracts in the database.
+            # This will tell us information about whether the applicant has been previously employed
+            offer_history =
+                Assignment.joins(:applicant, :offers, :position).where(
+                    applicant: applicant, "offers.status": 'accepted'
+                ).order("offers.position_start_date": :ASC).pluck(
+                    :"positions.position_code",
+                    :"offers.hours",
+                    :"offers.position_start_date",
+                    :"offers.position_end_date"
+                )
+            unless offer_history.blank?
+                data[:previous_department_ta] = true
+                data[:previous_university_ta] = true
+                # Technically we cannot infer this, but if they've TAed for the department
+                # before, we don't care about other history
+                data[:previous_other_university_ta] = false
```

Comment:
I don't really know where it's used so far, but just curious if we have to specify this value at all (would it be `true` if we didn't? or would it be an error if we didn't?)

Reply:
If we leave it off, it will be an unanswered question. Since it requires an answer, the user would have to interact with that page before moving on. Since UofT experience is way more important than any other experience, setting a default of `false` will prevent clutter in the applications.

Reply:
Oh I see, so we're prefilling false, makes sense!